include/revng/Yield/Assembly/DisassemblyHelper.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <map>
+#include <memory>
+
+#include "revng/Yield/Assembly/Assembly.h"
+
+namespace detail {
+class HiddenDisassemberStorage;
+}
+namespace model {
+class Function;
+}
+namespace efa {
+class FunctionMetadata;
+}
+class DisassemblerInstance;
+class RawBinaryView;
+
+class DissassemblyHelper {
+private:
+  detail::HiddenDisassemberStorage *Internal;
+
+public:
+  explicit DissassemblyHelper();
+  ~DissassemblyHelper();
+
+  assembly::Function disassemble(const model::Function &Function,
+                                 const efa::FunctionMetadata &Metadata,
+                                 const RawBinaryView &BinaryView);
+
+private:
+  DisassemblerInstance &access(MetaAddressType::Values AddressType);
```

Comment:
`access` -> `getInstance`?

Reply:
How about something like this https://github.com/revng/revng/pull/173/commits/c0975737d3057076daadb9c08ed0c007a4dfc479?

Reply:
`Specific` seems redundant, but OK.

Reply:
Mostly a hint at "thumb one is gonna be different" as in "it heavily depends on the type you pass in".

Reply:
There's no way to fetch a *non-specific* disassembler, no need to specify this.

Reply:
Why would anyone ever need a __non-specific__ disassembler though? You always want the one that can deal with the code before you.

I'll rename it though, it's no big deal.

Reply:
Done: https://github.com/revng/revng/pull/173/commits/68130aaa102516083aa1ca776d7cf60685c3ae8c.